protocol/protocol.go

```diff
@@ -288,12 +289,22 @@ func decodeFramed(req []byte) ([]byte, bool, error) {
 	return req, versionIETF, nil
 }
 
+func handleSRVTag(advertisedPreference []Version) bool {
+	for _, version := range advertisedPreference {
+		// The SRV tag is first defined in draft-ietf-ntp-roughtime-10
+		if version == VersionDraft10 {
+			return true
+		}
+	}
+	return false
+}
+
 // CreateRequest creates a Roughtime request given an entropy source and the
 // contents of a previous reply for chaining. If this request is the first of a
 // chain, prevReply can be empty. It returns the nonce (needed to verify the
 // reply), the blind (needed to prove correct chaining to an external party)
 // and the request itself.
-func CreateRequest(versionPreference []Version, rand io.Reader, prevReply []byte) (nonce, blind []byte, request []byte, err error) {
+func CreateRequest(versionPreference []Version, rand io.Reader, prevReply, rootPublicKey ed25519.PublicKey) (nonce, blind []byte, request []byte, err error) {
```

Comment:
nit: `prevReply` isn't a `ed25519.PublicKey`
```suggestion
func CreateRequest(versionPreference []Version, rand io.Reader, prevReply []byte, rootPublicKey ed25519.PublicKey) (nonce, blind []byte, request []byte, err error) {
```

Reply:
That was incredibly sloppy of me. I'll put up a PR to fix, one sec.

Reply:
https://github.com/cloudflare/roughtime/pull/56